packages/features/insights/server/routing-events.ts

```diff
@@ -454,6 +177,15 @@ class RoutingEventsInsights {
         utm_campaign: item.utm_campaign || "",
         utm_term: item.utm_term || "",
         utm_content: item.utm_content || "",
+        ...((bookingAttendees || [])
+          .filter(
+            (attendee): attendee is { name: string; email: string; timeZone: string | null } =>
+              typeof attendee.name === "string" && typeof attendee.email === "string"
+          )
+          .reduce((acc, attendee, index) => {
+            acc[`Attendee ${index + 1}`] = `${attendee.name} (${attendee.email})`;
+            return acc;
+          }, {} as Record<string, string>) || {}),
```

Comment:
```suggestion
...((bookingAttendees || [])
  .reduce((acc, attendee, index) => {
    if (typeof attendee.name === "string" && typeof attendee.email === "string") {
      acc[`Attendee ${index + 1}`] = `${attendee.name} (${attendee.email})`;
    }
    return acc;
  }, {} as Record<string, string>) || {})
```

Lets move the filter into the reduce directly so we don't have to iterate over the array twice

Reply:
yeah but actually what if

```
[
  { name: 'blah', email: 'hi@example' },
  { name: undefined, email: undefined },
  { name: 'hey', email: 'hey@example.com' }
]
```

then we will get

```
{
  "Attendee 1": "blah (hi@example)",
  "Attendee 3": "hey (hey@example.com)"
}
```

(We want Attendee "2" instead of 3)

we need to filter first, to get the correct "index" later.